src/backend/aspen/api/deps.py

```diff
@@ -55,3 +58,12 @@ def get_pathogen_slug(pathogen_slug=None):
     if pathogen_slug is None:
         return "SC2"
     return pathogen_slug
+
+
+async def get_pathogen(
+    slug: str = Depends(get_pathogen_slug), db: AsyncSession = Depends(get_db)
+) -> Pathogen:
+    try:
+        return await Pathogen.get_by_slug(db, slug)
+    except NoResultFound:
+        raise ex.BadRequestException("Invalid pathogen slug")
```

Comment:
A new dependency that checks that the pathogen slug in our URL is valid and returns an error if not.